src/test/java/emissary/util/ShortNameComparatorTest.java

```diff
@@ -37,6 +37,15 @@ public void testNonsenseNames() {
         assertEquals("Bad components sort in alpha order", "foo-att-abc", l.get(0).shortName());
     }
 
+    @Test
+    public void testNonsenseNames2() {
+        final List<IBaseDataObject> l = new ArrayList<IBaseDataObject>();
+        l.add(DataObjectFactory.getInstance(new Object[] {null, "def"}));
+        l.add(DataObjectFactory.getInstance(new Object[] {null, "abc"}));
+        Collections.sort(l, new ShortNameComparator());
+        assertEquals("Bad components sort in alpha order", "def", l.get(0).shortName());
+    }
+
     private void fillList(final List<IBaseDataObject> l) {
         l.add(DataObjectFactory.getInstance(new Object[] {this.nobytes, this.ba + "1"}));
```

Comment:
All of the items currently being added to this list contain `Family.SEP`.  Would it be useful to add at least 2 entries for which the shortname does not contain `Family.SEP` so we can assert the correct behavior when comparing those to each other and to the items that contain `Family.SEP`?